x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/CheckRolloverStep.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.indexlifecycle;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.admin.indices.rollover.RolloverInfo;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.unit.TimeValue;
+
+import java.util.function.LongSupplier;
+
+public class CheckRolloverStep extends AsyncWaitStep {
+    public static final String NAME = "check_rollover";
+
+    private static final Logger logger = LogManager.getLogger(CheckRolloverStep.class);
+    static final long TIMEOUT_MILLIS = TimeValue.timeValueMinutes(10).millis();
+
+
+    private LongSupplier nowSupplier;
+
+    CheckRolloverStep(StepKey key, StepKey nextStepKey, Client client, LongSupplier nowSupplier) {
```

Comment:
would it be better to minimize the places we introduce a dependency on System::currentTimeMillis and 
update `evaluateCondition` to have an additional parameter, LongSupplier nowSupplier?

I'm not saying this is necessarily better, was just curious about other's thoughts on minimizing external dependency entry points. It doesn't look like it matters for tests

Reply:
Given that this is the only step that needs it, I'm inclined to keep it in the constructor for now, and move it to an `evaluateCondition` parameter if we end up with more AsyncWaitSteps that need to know what time it is.  If anyone has strong feelings though, I would be open to promoting it to the AsyncWaitStep interface level.